pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -68,6 +68,18 @@ type DockerFilter struct {
 	Values []string `json:"values"`
 }
 
+// Filter is the configuration to limit the discovery process to a subset of available resources.
+type Filter struct {
```

Comment:
It might make sense to name it DockerSwarmFilter to be consistent with other structs. As a follow-up PR, we could consolidate DockerFilter and DockerSwarnFilter.